rest_framework/fields.py

```diff
@@ -227,16 +227,39 @@ def __init__(self, value, display_text, disabled=False):
         yield Option(value='n/a', display_text=cutoff_text, disabled=True)
 
 
+def _get_error_detail_dict(error_dict, default_code):
+    return {
+        k: [
+            ErrorDetail(error.message % (error.params or ()),
+                        code=error.code if error.code else default_code)
+            for error in errors
+        ] for k, errors in error_dict.items()
+    }
+
+
 def get_error_detail(exc_info):
     """
     Given a Django ValidationError, return a list of ErrorDetail,
     with the `code` populated.
     """
     code = getattr(exc_info, 'code', None) or 'invalid'
-    return [
-        ErrorDetail(msg, code=code)
-        for msg in exc_info.messages
-    ]
+
+    try:
+        error_dict = exc_info.error_dict
+    except AttributeError:
+        errors = []
+        for error in exc_info.error_list:
+            try:
+                error_dict = error.error_dict
+            except AttributeError:
+                detail = ErrorDetail(error.message % (error.params or ()),
+                                     code=error.code if error.code else code)
+            else:
+                detail = _get_error_detail_dict(error, code)
```

Comment:
This code/line is not executed.  Need to look into this, likely also wrapping it in a list, not sure if that's a valid use case though.